server/api/users/users.controller.ts

```diff
@@ -29,9 +29,32 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 };
 
 export const getUsers: RequestHandler<UserProfile[]> = async (req, res) => {
-  const users = await User.findAll({ attributes: User.allowedFields });
+  const page = req.query.page ? parseInt(req.query.page as string, 10) : 1;
+  const limit = 20;
+  const offset = (page - 1) * limit;
 
-  res.json(users);
+  if (!Number.isInteger(page) || page < 1) {
+    return res.status(400).json({ error: 'Invalid page number' });
+  }
+
+  // Fine all users, using the limit and offset to implement offset pagination
+  const users = await User.findAll({
+    attributes: User.allowedFields,
+    limit: limit,
+    offset: offset,
+  });
+
+  const count = await User.count();
+  const totalPages = Math.ceil(count / limit);
+
+  if (offset >= count && page !== 1) {
+    return res.status(400).json({ error: 'Page out of range' });
+  }
```

Comment:
I'd probably remove this check and just return an empty array along with the standard metadata... the client side can conditionally render based on the length of the returned `users` array.

Reply:
I don't believe I understand what you are requesting.